### PR TITLE
Clean up spec tests and pull in JJM 14149 #17

### PR DESCRIPTION
**Pull request by jerearista**

### CODE REVIEW

File: lib/puppet/provider/syslog_server/eos.rb [hunk 1]
```diff
@@ -40,11 +47,11 @@ def source_interface=(val)
 
   def create
     node.api('logging').add_host(resource[:name])
-    @provider_hash = { name: resource[:name], ensure: :present }
+    @property_hash = { name: resource[:name], ensure: :present }
```
Comment (devrobo) [inline lib/puppet/provider/syslog_server/eos.rb]: Why didn't the use of the wrong name break the unit test for the provider?

File: spec/unit/puppet/provider/snmp_notification/eos_spec.rb [hunk 2]
```diff
@@ -69,8 +69,8 @@ def snmp
 
       it { is_expected.to be_an Array }
 
-      it 'has only one entry' do
-        expect(subject.size).to eq 1
+      it 'has at least 1 entry' do
```
Comment (devrobo) [inline spec/unit/puppet/provider/snmp_notification/eos_spec.rb]: How come we don't know exactly how many entries are in the array?

File: lib/puppet/provider/port_channel/eos.rb [hunk 3]
```diff
@@ -59,7 +66,7 @@ def force=(val)
   end
 
   def speed=(val)
-    not_supported 'spped'
+    not_supported 'speed'
```
Comment (devrobo) [inline lib/puppet/provider/port_channel/eos.rb]: Does the broken code fail the unit test now?

File: lib/puppet/provider/syslog_server/eos.rb (hunk 1 above)
Comment (jerearista) [inline lib/puppet/provider/syslog_server/eos.rb]: changing "provider_hash" to "property_hash" fixes the following failure:

      1) Puppet::Type::Syslog_server::ProviderEos resource (instance) methods #create sets ensure to :present
     Failure/Error: expect(provider.ensure).to eq(:present)
       
       expected: :present
            got: :absent
       
       (compared using ==)
       
       Diff:
       @@ -1,2 +1,2 @@
       -:present
       +:absent
     # ./spec/unit/puppet/provider/syslog_server/eos_spec.rb:141:in `block (4 levels) in <top (required)>'

File: spec/unit/puppet/provider/snmp_notification/eos_spec.rb (hunk 2 above)
Comment (jerearista) [inline spec/unit/puppet/provider/snmp_notification/eos_spec.rb]: That was left over from sorting out some test fixtures.   There are 34, currently, based on the default data from EOS:

    grep name: spec/fixtures/fixture_eapi_snmp.yaml | wc -l
      34

File: lib/puppet/provider/port_channel/eos.rb (hunk 3 above)
Comment (jerearista) [inline lib/puppet/provider/port_channel/eos.rb]: No.  There currently are no tests for the NotImplemented exception.

File: spec/unit/puppet/provider/snmp_notification/eos_spec.rb (hunk 2 above)
Comment (jerearista) [inline spec/unit/puppet/provider/snmp_notification/eos_spec.rb]: Resolved in commit 05cb83cd650322cc4ef1fc7b91c4a92c8ddad249

**Pull request merged**